pkg/aws/actuator/actuator.go

```diff
@@ -651,12 +651,25 @@ func (a *AWSActuator) awsPolicyEqualsDesiredPolicy(desiredUserPolicy string, aws
 	return true, nil
 }
 
-func userHasExpectedTags(logger log.FieldLogger, user *iam.User, infraName, clusterUUID string) bool {
+func userHasExpectedTags(logger log.FieldLogger, user *iam.User, infraName, clusterUUID string, awsClient client.Client) bool {
 	// Check if the user has the expected tags:
 	if user == nil {
 		return false
 	}
 
+	// Check if the user has the expected tags from the Infrastructure resource
+	infra, _ := utils.GetInfrastructure(awsClient)
+	if infra != nil {
+		if len(infra.Spec.PlatformSpec.AWS.ResourceTags) != 0 {
+			for _, userTag := range infra.Spec.PlatformSpec.AWS.ResourceTags {
+				if !userHasTag(user, userTag.Key, userTag.Value) {
+					log.Warnf("user missing tag: %s=%s", userTag.Key, userTag.Value)
```

Comment:
Is a missing tag an unexpected situation? If not, then perhaps a log.Info/Debug() would be more appropriate.

Reply:
Agreed.